builder/upcloud/config.go

```diff
@@ -51,24 +52,54 @@ type Config struct {
 	common.PackerConfig `mapstructure:",squash"`
 	Comm                communicator.Config `mapstructure:",squash"`
 
-	// Required configuration values
-	Username    string `mapstructure:"username"`
-	Password    string `mapstructure:"password"`
-	Zone        string `mapstructure:"zone"`
-	StorageUUID string `mapstructure:"storage_uuid"`
-	StorageName string `mapstructure:"storage_name"`
+	// The username to use when interfacing with the UpCloud API.
+	Username string `mapstructure:"username" required:"true"`
 
-	// Optional configuration values
-	TemplatePrefix string        `mapstructure:"template_prefix"`
-	TemplateName   string        `mapstructure:"template_name"`
-	StorageSize    int           `mapstructure:"storage_size"`
-	Timeout        time.Duration `mapstructure:"state_timeout_duration"`
-	CloneZones     []string      `mapstructure:"clone_zones"`
+	// The password to use when interfacing with the UpCloud API.
+	Password string `mapstructure:"password" required:"true"`
 
+	// The zone in which the server and template should be created (e.g. nl-ams1).
+	Zone string `mapstructure:"zone" required:"true"`
+
+	// The UUID of the storage you want to use as a template when creating the server.
+	StorageUUID string `mapstructure:"storage_uuid" required:"true"`
+
+	// The name of the storage that will be used to find the first matching storage in the list of existing templates.
+	//
+	// Note that `storage_uuid` parameter has higher priority. You should use either `storage_uuid` or `storage_name` for not strict matching (e.g "ubuntu server 20.04").
+	StorageName string `mapstructure:"storage_name" required:"true"`
+
+	// The prefix to use for the generated template title.
+	// Defaults to an empty string, meaning the prefix will be the storage title.
+	// You can use this option to easily differentiate between different templates.
+	TemplatePrefix string `mapstructure:"template_prefix"`
```

Comment:
I think this could be refined a bit. By default the prefix is defined by a const `DefaultImagePrefix` which currently is `custom-image`. Looks like there's also an existing issue about a similar inconsistency #15

Reply:
Good point! I'll update the description.